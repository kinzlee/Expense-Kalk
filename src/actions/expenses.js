import uuid from 'uuid';

// ACTION GENERATOR for expense  reducer

export const addExpenses = ({
    description = '',
    messageNote = '',
    amount = 0,
    createdAt = 0
} = {}) => {
    return {
        type : 'ADD_EXPENSES',
        expenses : {
            id: uuid(),
            description,
            messageNote,
            amount,
            createdAt
        }
    }
};

export const removeExpense = ({id} = {}) => {
    return {
        type : 'REMOVE_EXPENSE',
        id
    }
};



export const editExpense = (id, update) => {
    return {
        type : 'EDIT_EXPENSE',
        id,
        update
    }
};