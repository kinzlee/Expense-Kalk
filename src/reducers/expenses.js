// Expenses Reducer
const defaultExpenseState = [];

export default (state=defaultExpenseState, action) => {
    switch(action.type) {
        
        case  'ADD_EXPENSES':
            return [
            ...state,
            action.expenses
            ];

        case 'REMOVE_EXPENSE' : 
            return state.filter( ({id}) => id !== action.id);
            
        case 'EDIT_EXPENSE' :
            return state.map((expense) => {
                if(expense.id === action.id) {
                    return {
                        ...expense,
                        ...action.update
                    }
                }
                    return expense
            });

        default : 
        return state

           
    }
};