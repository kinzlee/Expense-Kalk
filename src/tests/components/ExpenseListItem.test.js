import React from 'react';
import {shallow} from 'enzyme';
import ExpensesListItem from '../../components/ExpenseListItem';
import expenses from '../fixture/expenses';


test('should render ExpenseList with expenses', () => {
    const wrapper = shallow(<ExpensesListItem expenses={expenses[1]}/>);
    expect(wrapper).toMatchSnapshot(); 
});