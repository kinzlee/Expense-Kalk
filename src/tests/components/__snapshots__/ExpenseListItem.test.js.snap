// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render items of the ExpenseList with expenses 1`] = `
<div>
  <Link
    to="/edit/2"
  >
    <h2>
       
      phone
       
    </h2>
  </Link>
  <p>
    45,000.00
    -
    1st, January 1970
  </p>
</div>
`;
